### PR TITLE
Namespace properties

### PR DESCRIPTION
**Pull request by sreevatsanraman**
* Added properties for namespace

### CODE REVIEW

**Comment by sreevatsanraman** [comment]
Build: http://builds.cask.co/browse/CDAP-DUT1015-1

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/NamespaceAdmin.java [hunk 1]
```diff
@@ -71,4 +70,13 @@
...
   public void deleteNamespace(Id.Namespace namespaceId) throws NotFoundException, NamespaceCannotBeDeletedException;
+
+  /**
+   * Update namespace properties for a given namespace.
+   *
+   * @param namespaceId  the {@link Id.Namespace} of the namespace to be updated
+   * @param properties  Map of properties to be updated
+   * @throws NotFoundException if the specified namespace is not found
+   */
+  public void updateProperties(Id.Namespace namespaceId, Map<String, String> properties) throws NotFoundException;
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/NamespaceAdmin.java]: Should we also allow update for name and description as part of this (not this API, but this PR)? We have [CDAP-1431](https://issues.cask.co/browse/CDAP-1431) open for it.

Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/NamespaceAdmin.java]: Maybe the create API could update name and description if the namespace exists, else create a new one?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java [hunk 2]
```diff
@@ -797,6 +797,21 @@ public NamespaceMeta apply(AppMds input) throws Exception {
   }
 
   @Override
+  public void updateNamespace(final NamespaceMeta metadata) {
+    Preconditions.checkArgument(metadata != null, "Namespace metadata cannot be null.");
+    txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, Void>() {
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java]: Should we start throwing checked exceptions from the ``Store``? Or keep it consistent right now and fix it all later?

**Comment by bdmogal** [comment]
Just a couple of questions.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java (hunk 2 above)
Comment (sreevatsanraman) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java]: Let's keep this consistent and refactor at once. There is a JIRA open for cleaning up exception handling across CDAP

**Comment by anwar6953** [comment]
LGTM

**Comment by sreevatsanraman** [comment]
Will merge after this build passes: http://builds.cask.co/browse/CDAP-DUT1015-2

**Comment by sreevatsanraman** [comment]
Unrelated build failure. Updated the [CDAP-1683](https://issues.cask.co/browse/CDAP-1683) with the info.

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java [hunk 3]
```diff
@@ -47,6 +52,8 @@
 @Path(Constants.Gateway.API_VERSION_3)
 public class NamespaceHttpHandler extends AbstractAppFabricHttpHandler {
   private static final Logger LOG = LoggerFactory.getLogger(NamespaceHttpHandler.class);
+  private static final Gson GSON = new Gson();
+  private static final Type STRING_MAP_TYPE = new TypeToken<Map<String, String>>() { }.getType();
```
Comment (gsps1) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java]: unused ?

Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java]: @gsps1 will remove in an upcoming PR.

**Pull request merged**